c2rust-analyze/src/main.rs

```diff
@@ -361,6 +361,53 @@ impl<'tcx> Debug for AdtMetadataTable<'tcx> {
     }
 }
 
+/// Determine if a [`Constant`] is a (byte-)string literal.
+///
+/// The current implementation is super hacky and just uses [`Constant`]'s [`Display`] `impl`,
+/// but I'll soon replace it with a more robust (but complex) version
+/// using pretty much the same way the [`Display`] `impl` does it.
+///
+/// [`Display`]: std::fmt::Display
+fn is_string_literal(c: &Constant) -> bool {
+    let s = c.to_string();
+    s.ends_with('"') && {
+        let s = match s.strip_prefix("const ") {
+            Some(s) => s,
+            None => return false,
+        };
+        s.starts_with('"') || s.starts_with("b\"")
+    }
```

Comment:
This really is horrible, but the predicate is well-defined so a different implementation should be easy to slot in.

Reply:
Yeah, it's unfortunate.  I think this actually also has a lower chance of breaking in future `rustc` versions than an implementation based on what the `Display` `impl` does.